__tests__/integration/block-finder-incremental.test.ts

```diff
@@ -0,0 +1,163 @@
+      await provider.destroy();
+    }
+  });
+
+  describe("Incremental processing after interruption", () => {
+    it("should resume processing after interruption without duplicate RPC calls", async () => {
+      // Test data: 5-day range from January 9-13, 2024
+      // Using dates from the test data referenced in the issue
+      const startDate = new Date("2024-01-09");
+      const endDate = new Date("2024-01-13");
+
+      // Phase 1: Start processing and interrupt after 2 days
+      // First, we'll process only Jan 9-10 (2 days of the 5-day range)
+      const partialEndDate = new Date("2024-01-10");
+      const expectedPartialDays = ["2024-01-09", "2024-01-10"];
+
+      console.log("Phase 1: Processing first 2 days (Jan 9-10)...");
+      const initialRpcCallCount = rpcCallCount;
+
+      const partialResult = await blockFinder.findBlocksForDateRange(
+        startDate,
+        partialEndDate,
+      );
+
+      // Verify we found blocks for the first 2 days
+      expectedPartialDays.forEach((date) => {
+        expect(partialResult.blocks).toHaveProperty(date);
+        expect(partialResult.blocks[date]).toBeGreaterThan(0);
+      });
+      expect(Object.keys(partialResult.blocks)).toHaveLength(
+        expectedPartialDays.length,
+      );
+
+      const rpcCallsForFirstTwoDays = rpcCallCount - initialRpcCallCount;
+      console.log(
+        `RPC calls made for first 2 days: ${rpcCallsForFirstTwoDays}`,
+      );
+      expect(rpcCallsForFirstTwoDays).toBeGreaterThan(0);
+
+      // Simulate interruption by creating a new BlockFinder instance
+      // This represents the process restarting after a failure
+      console.log("Simulating interruption and restart...");
+
+      const newProvider = createProvider();
+      const { getCallCount } = instrumentProviderForCallTracking(newProvider);
+      const resumedBlockFinder = new BlockFinder(
+        testContext.fileManager,
+        newProvider,
+      );
+
+      // Phase 2: Resume processing for the full date range
+      console.log("Phase 2: Resuming for full range (Jan 9-13)...");
+
+      const fullResult = await resumedBlockFinder.findBlocksForDateRange(
+        startDate,
+        endDate,
+      );
+
+      // Verify we have all 5 days
+      const expectedAllDays = [
+        "2024-01-09",
+        "2024-01-10",
+        "2024-01-11",
+        "2024-01-12",
+        "2024-01-13",
+      ];
+
+      expectedAllDays.forEach((date) => {
+        expect(fullResult.blocks).toHaveProperty(date);
+        expect(fullResult.blocks[date]).toBeGreaterThan(0);
+      });
+      expect(Object.keys(fullResult.blocks)).toHaveLength(
+        expectedAllDays.length,
+      );
+
+      // Critical assertion: Verify no RPC calls were made for already-found blocks
+      // We should only have made calls for the 3 remaining days (Jan 11-13)
+      const resumeRpcCallCount = getCallCount();
+      console.log(`RPC calls made during resume: ${resumeRpcCallCount}`);
```

Comment:
We arent verifying the number of rpc calls here - only logging it. Same comment for above where we get the call count. I guess this is because we're doing a binary search here. Perhaps a better way of verifying the behaviour would be to check that none of the blocks are requested within the date range that we're not interested in, and instead they're all within the date range (or date range + 1) that we are intereseted in.